### ISSUE TITLE
Support subheadings as content for Back of a card

### ISSUE THREAD

**Issue by kenhuang-io** [eyeinsky/org-anki#66]
Hi,

Thanks for this nice package!

I noticed that currently it doesn't support to include subheadings as content for Back. For example, Back content for `test` only contains "aaa", without content of the `sub` subheading:

```
** test
aaa
*** sub
bbb
```

Do you think it's a good idea to support this? Thanks.

**Comment by eyeinsky** [comment]
Hi, yes, this would be a useful feature. I guess the method would be to add a special property for it which one could add to the heading, at the file level and then as a global customizable variable.

There would be a minor difficulty for the command that syncs all headings in a file -- it would need to start skipping the subheadings that are already added as part of their parent heading.

**Comment by kenhuang-io** [comment]
Hi Markus,

I'm not sure what a typical heading hierarchy looks like for =org-anki-sync-all=. For me, I'd like to put all node headings at the same level in a org-mode file, so I'm wondering maybe we could define a node level property so that org-anki-sync-all and org-anki-sync/delete-entry would know what headings are interested. (P.S. I have a command named org-anki-sync-subentries locally for this hierarchy.)

Or more flexibly, we could define a "node container" property or something like that, so that all those commands will only consider the subheadings inside them as nodes.

What do you think?

**Comment by eyeinsky** [comment]
Hi, and thank you for the explanation, now I understand the issue! :)

How `org-anki-sync-all` currently works is that iterates through all entries _regardless of what their level is_ and makes cards from them (see below for example[1]). Both of the solutions you mention can theoretically work (1: have a fixed level, 2: have a entry property where all subheadings are taken as content, not as new cards). Just need to figure out what in the org-mode API can help us implement these. 

Maybe you can also share your `org-anki-sync-subentries` somewhere to see how it's implemented?

[1] All of these will become separate cards:
```org-mode
* Card 1 title
Card 1 content
** Card 2 title
Card 2 content
*** Card 3 title
Card 3 content
```

**Comment by kenhuang-io** [comment]
Hi,

For me, it's more likely to start a new first level heading dedicated for Anki cards in an existed org file, and then add second level headings as cards. So I need a syncing subentries command, it's like this:

``` elisp
(defun w/org-anki-sync-subentries ()
  "Sync all subentries of the current entry at point.

Also respect `org-anki-skip-function'."
  (interactive)
  (save-excursion
    (let* ((next-level (1+ (org-current-level))))
      (org-map-entries (lambda ()
                         (when (and t
                                    ;; (member "foo" (org-get-tags)) ; ad-hoc tag filtering
                                    (= next-level (org-element-property :level (org-element-at-point))))
                           (org-anki-sync-entry)))
                       nil
                       'tree
                       org-anki-skip-function))))
```

**Comment by basaran** [comment]
Hello, thank you Ken. I slightly modified your `subentries` idea to only create cards from a fixed level. Meanwhile, this might be unorg proposal, but I have been thinking since last night for a different matter.

Perhaps, it could also make sense to use an additional condition inside `org-anki--get-fields`. Please let me add a screenshot of the layout.

Given that we have a new note defined in Anki, such as:

```
(use-package! org-anki
   :config
   (add-to-list 'org-anki-model-fields '("Typewriter" "Front" "Back", "Explain")))
```

![image](https://github.com/eyeinsky/org-anki/assets/30809170/233c24c0-b2d3-4e61-894a-7488637a464e)

--

### But Why?

I understand this may seem unnecessary, but the issue I ran into was with Anki's {{FrontSide}} field, which is used with the {{type:Back}}. It allows you to do something like this:

![image](https://github.com/eyeinsky/org-anki/assets/30809170/48fb922f-0e59-45a4-ae28-f52743103867)


If I had a note such as this:

![image](https://github.com/eyeinsky/org-anki/assets/30809170/56f17904-f1ce-4d8e-9dd0-8d24b8d0d3f4)

Then the answer would look like this:

![image](https://github.com/eyeinsky/org-anki/assets/30809170/33df851d-03d1-44ec-9ca1-034b92aae48a)

But, it would be preferable to have it look like this:

![image](https://github.com/eyeinsky/org-anki/assets/30809170/5571c100-1e6f-40da-9514-f84ada78879f)

### Summary
For this specific instance, it would help to be able to isolate the Back from the rest of the things.

And it could be helpful to come up with a few additional conditions inside the --get-fields function, so we don't need to populate full structures like you would do with anki-editor. Hope this makes sense and is in line with Ken's comments.

**Comment by eyeinsky** [comment]
@basaran Could you create a new issue for this, so we don't mix up the two :)

**Comment by eyeinsky** [comment]
@whatacold I'm thinking of implementing this through a special command which does the following two things
- add a skip function to include entries at a fixed depth
- add a setting to enable syncing of sub-headings

Why have a special command: the second point above doesn't make sense when the first one isn't set (users would be shooting themselves in the foot by sync all subcontent for all org entries recursively).

**Comment by basaran** [comment]
> @basaran Could you create a new issue for this, so we don't mix up the two :)

I know, I'm sorry. Just wanted to provide the details to the summary but it ended up going in a different direction. I will create the thread shortly.

**Comment by v-Nyo** [comment]
This would also allow compatibility with Logseq flashcards (supports .org, and org-roam). I would really like the option to export my 10000+ Logseq flashcards to anki without any headache.

**Comment by v-Nyo** [comment]
How would org-anki handle heading tags? In Logseq a flash card looks like this:

* FRONT0 :card:
* * BACK0
* Front1 :card:English:
* *BACK2
* FRONT2 :card:Spanish:
* * BACK2
* FRONT3 :card:
* * BACK3 

like the most simple org-drill card. But instead of drill tag the card tag. Subheadings don't inherit tags in Logseq

**Comment by v-Nyo** [comment]
This would be awesome and also provide instant compatibility with Logseq (:card: tag).

**Comment by v-Nyo** [comment]
Notably pls let it also include content from under the subheadings too. Like this:

```
* FRONT CONTENT      :card:
  FRONT CONTENT
* * BACK CONTENT
    BACK CONTENT
```

Honestly as a Logseq user, I really need this because the current SRS algorithm is faulty (FSRS will still take a while, if ever). I wish I could actually write elisp so that I could send a PR. (I don't get good results when I export from Logseq to anki, not to mention that it also breaks org-roam)

**Comment by eyeinsky** [comment]
@v-Nyo I'm not familiar with Logseq -- does it support syncing to org-mode?

**Comment by v-Nyo** [comment]
@eyeinsky Sorry should have explained more. Logseq is a plaintext knowledge management system like org-roam. It supports markdown AND org-mode. In Logseq I just need to select my org-roam dir and everything just works (with minimal configuration in emacs https://coredumped.dev/2021/05/26/taking-org-roam-everywhere-with-logseq/)

**Comment by v-Nyo** [comment]
I am using Logseq on the go (emacs on android and emacs in termux don't work for me). But I basically write everything in emacs and just use Logseq to view my notes and do flashcards + a couple other things. 

Logseq also stores some flashcard stuff inside the drawer of the heading. A card looks like this:

```
* How do you create a paragraph in html? :card:webdesign:
:PROPERTIES:
:card-last-interval: 1
:card-repeats: 1
:card-ease-factor: 2.36
:card-next-schedule: 2023-12-08T10:27:30.737Z
:card-last-reviewed: 2023-12-07T10:27:30.738Z
:card-last-score: 3
:END:
** You use the <p> element. here is an example:
#+begin_src html
<p> Hi </p>
#+end_src

```
I want to switch to anki because it already supports FSRS (Ankidroid will in the next release) and Logseq has the issues mentioned above.

**Comment by v-Nyo** [comment]
Hey, do you still want to implement this?

In about 4 Months I might finally have some time to learn elisp.
I still really want this feature, so I would appreciate some advice on where to start / how best to go about this

**Comment by eyeinsky** [comment]
Yes, this feature still makes sense. I haven't currently had time to implement it or even look at how to do it :(, but if you could help, that would be great!

If you're just starting out with elisp, then a way forward could be that you create a branch in git, then add some stub content to it, then push it and create a draft pull request against this repo -- then in the pull request comments we could do the whole development.

A good first target would be to add a function `org-anki-sync-subtree` which would create a card where org entry title would be the Front and the entire content (including subheadings) would be the card Back.
